### PR TITLE
Add `parallelMap`

### PR DESCRIPTION
**Pull request by freekmurze**
This PR adds a `parallelMap` macro.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -29,7 +29,12 @@
         "illuminate/view": "~5.5.0"
     },
     "require-dev": {
-        "amphp/parallel-functions": "^0.1.0",
+        "amphp/amp": "^2.0.4",
+        "amphp/byte-stream": "^1.2.1",
+        "amphp/parallel": "^0.2.0",
+        "amphp/parallel-functions": "^0.1.1",
+        "amphp/parser": "^1.0.0",
+        "amphp/process": "^0.3.1",
```
Comment (kelunik) [inline composer.json]: Why that?

Comment (kelunik) [reply composer.json]: You just need `"amphp/amp": "^2.0.3"` here, as that fixed the predictable order of combinators such as `Amp\Promise\all()`.

Comment (kelunik) [reply composer.json]: Please depend on `amphp/parallel-functions@^0.1.2` instead.

Comment (freekmurze) [reply composer.json]: This was just a first stab at getting it to work with the lower recommendations. 

I've changed this, now it follows your recommendations. 

Thanks!

File: composer.json [hunk 2]
```diff
@@ -23,16 +23,22 @@
...
     "require": {
         "php": "^7.0",
+        "illuminate/pagination": "~5.5.0",
         "illuminate/support": "~5.5.0",
         "illuminate/validation": "~5.5.0",
-        "illuminate/pagination": "~5.5.0",
         "illuminate/view": "~5.5.0"
     },
     "require-dev": {
+        "amphp/amp": "^2.0.3",
```
Comment (kelunik) [inline composer.json]: You can drop that one now, as `amphp/parallel-functions@0.1.2` enforces that.

Comment (freekmurze) [reply composer.json]: Will do

**Pull request merged**